dht_options.go

```diff
@@ -67,6 +69,23 @@ func (c *config) apply(opts ...Option) error {
 	return nil
 }
 
+// applyFallbacks sets default values that could not be applied during config creation since they are dependent
+// on other configuration parameters (e.g. optA is by default 2x optB) and/or on the Host
+func (c *config) applyFallbacks(h host.Host) {
+	if c.validator == nil {
+		if c.protocolPrefix == DefaultPrefix {
+			c.validator = record.NamespacedValidator{
+				"pk":   record.PublicKeyValidator{},
+				"ipns": ipns.Validator{KeyBook: h.Peerstore()},
+			}
+		} else {
+			c.validator = record.NamespacedValidator{
+				"pk":   record.PublicKeyValidator{},
+			}
```

Comment:
We've had this in in the past, but I'm not quite sure why... does anyone even use this?